### PR TITLE
feat(autolayout): collapsed children don't count in the measure

### PR DESCRIPTION
**Pull request by Robert-Louis**
GitHub Issue (If applicable): #660

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

Collapsed children are still measure

## What is the new behavior?

Collapsed children are not measure

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [ ] [Docs](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc) have been added/updated
- [ ] [Runtime Tests and/or UI Tests](https://platform.uno/docs/articles/contributing/guidelines/creating-tests.html) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal)
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by agneszitte — COMMENTED** [review]
@Robert-Louis Can you add some tests for these changes, please
To make sure when an element is collapsed in AutoLayout it will not take any more space

File: src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Arrange.cs [hunk 1]
```diff
@@ -47,6 +47,11 @@ protected override Size ArrangeOverride(Size finalSize)
 			var child = children[i];
 			var calculatedChild = _calculatedChildren[i];
 
+			if (child as FrameworkElement is { } fe && fe.Visibility is Visibility.Collapsed)
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Arrange.cs:50]:
```suggestion
			if (child as FrameworkElement is { Visibility: Visibility.Collapsed })
```

Comment (kazo0) [inline src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Arrange.cs:50]:
```suggestion
			if (child is FrameworkElement { Visibility: Visibility.Collapsed } fe)
```

File: src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Arrange.cs [hunk 2]
```diff
@@ -152,6 +157,11 @@ protected override Size ArrangeOverride(Size finalSize)
 				continue; // next child, current offset remains unchanged
 			}
 
+			if (children[i] as FrameworkElement is { } fe && fe.Visibility is Visibility.Collapsed)
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Arrange.cs:160]:
```suggestion
			if (children[i] as FrameworkElement is { Visibility: Visibility.Collapsed })
```

Comment (kazo0) [inline src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Arrange.cs:160]:
```suggestion
			if (children[i] is FrameworkElement { Visibility: Visibility.Collapsed } fe)
```

**Comment by Robert-Louis** [comment]
> @Robert-Louis Can you add some tests for these changes, please To make sure when an element is collapsed in AutoLayout it will not take any more space

I test it manualy. Unfortunatly I need to go now so I will not have the time add it today.

File: src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Measure.cs [hunk 3]
```diff
@@ -212,7 +213,7 @@ private void ApplyMinMaxValues(ref Size desiredSize)
 		{
 			var calculatedChild = _calculatedChildren[i];
 
-			if (calculatedChild.Role != AutoLayoutRole.Fixed)
+			if (calculatedChild.Role != AutoLayoutRole.Fixed && !calculatedChild.IsVisible)
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Measure.cs:216]: should the logic here be && or ||?

**Review by Xiaoy312 — APPROVED** [review]

**Comment by agneszitte** [comment]
> @kazo0 / @Xiaoy312 @Robert-Louis will be OOO for 10 days starting at noon today, so additional adjustments will need to be picked up by someone else if we want to have the changes merged soon

@Robert-Louis you will be able to work on the tests when you are back, please
We will merge the current changes to at least have the fix during that time

File: src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Arrange.cs [hunk 4]
```diff
@@ -47,6 +47,11 @@ protected override Size ArrangeOverride(Size finalSize)
 			var child = children[i];
 			var calculatedChild = _calculatedChildren[i];
 
+			if (child is FrameworkElement { Visibility: Visibility.Collapsed })
```
Comment (jeromelaban) [inline src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Arrange.cs:50]: @Robert-Louis  there should have been a runtime time for this change. Was it possible to create one?

**Comment by agneszitte** [comment]
https://github.com/Mergifyio backport release/stable/3.0

**Comment by agneszitte** [comment]
https://github.com/Mergifyio backport legacy/3x

**Comment by agneszitte** [comment]
> > @kazo0 / @Xiaoy312 @Robert-Louis will be OOO for 10 days starting at noon today, so additional adjustments will need to be picked up by someone else if we want to have the changes merged soon
> 
> @Robert-Louis you will be able to work on the tests when you are back, please We will merge the current changes to at least have the fix during that time

Related PR for the Tests here : https://github.com/unoplatform/uno.toolkit.ui/pull/691
(cc @jeromelaban / @kazo0 / @Xiaoy312

**Pull request merged**